src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2524,9 +2545,14 @@ def load_key_and_certificates_from_pkcs12(self, data, password):
         if sk_x509_ptr[0] != self._ffi.NULL:
             sk_x509 = self._ffi.gc(sk_x509_ptr[0], self._lib.sk_X509_free)
             num = self._lib.sk_X509_num(sk_x509_ptr[0])
+
             # In OpenSSL < 3.0.0 PKCS12 parsing reverses the order of the
             # certificates.
-            for i in reversed(range(num)):
+            indices: typing.Iterable[int] = range(num)
+            if not self._lib.CRYPTOGRAPHY_OPENSSL_300_OR_GREATER:
+                indices = reversed(range(num))
```

Comment:
Can you either write the second line as `indicies = reversed(indiced)` or write the first assignment in an `else`? I don't like the assignment that's dead if you follow the second path.